packages/api/src/libs/utils.ts

```diff
@@ -83,3 +83,49 @@ export function msToS(ms: number) {
 export function nowS(): number {
   return msToS(Date.now());
 }
+
+// utility to help convert multicall responses to string, these are some times in nested arrays
+export function toString(x: any) {
+  if (lodash.isArray(x)) {
+    return x[0].toString();
+  }
+  return x.toString();
+}
+// utility to help convert multicall responses to number, these are some times in nested arrays
+export function toNumber(x: any) {
+  if (lodash.isArray(x)) {
+    return Number(x[0]);
+  }
+  return Number(x);
+}
+// utility to help convert multicall responses of bytes32 to string, these are some times in nested arrays
+export function parseBytes(x: any) {
+  if (lodash.isArray(x)) {
+    return parseBytes32String(x[0]);
+  }
+  return parseBytes32String(x);
+}
+
+export const BatchRead = (multicall: uma.Multicall) => async (
+  calls: [string, (x: any) => any][],
+  contract: Contract
+) => {
+  // multicall batch takes array of {method} objects
+  const results = await multicall
+    .batch(
+      contract,
+      calls.map(([method]) => ({ method }))
+    )
+    .read();
+
+  // convert results of multicall, an array of responses, into a key value, keyed by contract method
+  return Object.fromEntries(
+    lodash.zip(calls, results).map((zipped) => {
+      const [method, result] = zipped;
+      if (method == null) return [];
+      if (result == null) return [];
+      const [key, map] = method;
+      return [key, map(result)];
+    })
+  );
+};
```

Comment:
this file was update with multicall helper functions. the multicall library in SDK is really bare bones, these helpers take care of parsing results to string/numbers and putting the results back into a key/value object. the multicall library doesnt do this and just returns a raw array of results.